objects/src/chain/mod.rs

```diff
@@ -1,39 +1,79 @@
-use super::{crypto::merkle::Mmr, AdviceInputsBuilder, ToAdviceInputs};
+use super::{
+    crypto::merkle::{MmrPeaks, PartialMmr},
+    utils::collections::{BTreeMap, Vec},
+    AdviceInputsBuilder, Digest, ToAdviceInputs,
+};
 
-// TODO: Consider using a PartialMmr that only contains the Mmr nodes that are relevant to the
-// transaction being processed.
+// CHAIN MMR
+// ================================================================================================
 
-/// A struct that represents the chain Mmr accumulator.
-/// This wraps the [Mmr] object and provides a simple interface to access / modify it.
-/// We use a custom type here as the traits we implement on this type could be context specific.
+/// A struct that represents the chain Merkle Mountain Range (MMR).
 ///
-/// The Mmr allows for efficient authentication of consumed notes during transaction execution.
-/// Authentication is achieved by providing an inclusion proof for the consumed notes in the
-/// transaction against the chain Mmr root associated with the latest block known at the time
+/// The MMR allows for efficient authentication of input notes during transaction execution.
+/// Authentication is achieved by providing inclusion proofs for the notes consumed in the
+/// transaction against the chain MMR root associated with the latest block known at the time
 /// of transaction execution.
-#[derive(Clone, Default, Debug)]
-#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct ChainMmr(Mmr);
```

Comment:
Will we ever need to instantiate from an `Mmr`? If so, we could have another constructor `from(mmr: Mmr)`, and make the `ChainMmr` an `enum`.

This would remove the need for `mock::chain::mmr_to_chain_mmr()` function

Reply:
I think we might need this down the road, but the implementation will be different from what `mock::chain::mmr_to_chain_mmr()` is doing. This method is suitable only for tests as it is basically dumping the whole `Mmr` into a `PartialMmr`. What we'll probably need is to instantiate a partial MMR from a subset of a full MMR. So, the function could look something like:
```Rust
pub fn from_mmr(mmr: &Mmr, positions: &[usize]) -> ChainMmr
```
But I also haven't thought this through enough - so, decided to leave for the future.

Reply:
Makes sense